### PR TITLE
try solve the numbers equations

### PR DESCRIPTION
**Pull request by omar92**
if the number input is something like 2+(3*2)
instead of writing Nan

it will be solved 

8 will be written

### CODE REVIEW

**Comment by atlasan** [comment]
I'm not a big fan of this, what are trying to achieve @omar92 ?
A formula node would probably serve this better, no?

**Comment by omar92** [comment]
Thanks for feedback , sadly   LGraphNode.prototype.getInputOrProperty occurs after the data is converted to Nan 

will try specific widget thing , but it is confusing to me, I'm a c# dev XD so still trying to figure this out in python and JS

**Comment by omar92** [comment]
ok will work on it thanks so much for the help

File: src/litegraph.js [hunk 1]
```diff
@@ -9985,6 +9985,12 @@ LGraphNode.prototype.executeAction = function(action)
 						var delta = x < 40 ? -1 : x > widget_width - 40 ? 1 : 0;
 						if (event.click_time < 200 && delta == 0) {
 							this.prompt("Value",w.value,function(v) {
+									// check if v is a valid equation or a number
```
Comment (atlasan) [inline src/litegraph.js:9988]: Could check `if(LiteGraph.allow_scripts){`

**Comment by jagenjo** [comment]
using eval is very dangerous, I see that you put it behind a regex to ensure some subset of characters, but still, you are leaving all the secutiry check into one regex and somebody may find a way to bypass it. 

Although it is true that this will mean the bad stuff is executed locally at your own risk, which is more or less like opening the console and typing some JS, so it is not so dangerous. I will think about it.

**Comment by jagenjo** [comment]
Anyway, lets accept it, it could be useful after all.

**Pull request merged**